src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol

```diff
@@ -0,0 +1,451 @@
+    {
+        // Convert depositAmount to 18 decimals, which is required by Bancor formula
+        uint decimalConvertedDepositAmount = _convertAmountToRequiredDecimal(
+            _depositAmount,
+            IERC20MetadataUpgradeable(address(token())).decimals(),
+            eighteenDecimals
+        );
+        // Calculate mint amount through bonding curve
+        mintAmount = formula.calculatePurchaseReturn(
+            virtualTokenSupply,
+            virtualCollateralSupply,
+            reserveRatioForBuying,
+            decimalConvertedDepositAmount
+        );
+        // Convert mint amount to issuing token decimals
+        mintAmount = _convertAmountToRequiredDecimal(
+            mintAmount, eighteenDecimals, decimals()
+        );
+    }
+
+    /// @dev Calculates the amount of collateral to be received when redeeming a given amount of tokens.
+    /// This internal function is an override of RedeemingBondingCurveFundingManagerBase's abstract function.
+    /// It handles decimal conversions and calculations through the bonding curve.
+    /// @param _depositAmount The amount of tokens to be redeemed for collateral.
+    /// @return redeemAmount The amount of collateral that will be received.
+    function _redeemTokensFormulaWrapper(uint _depositAmount)
+        internal
+        view
+        override(RedeemingBondingCurveFundingManagerBase)
+        returns (uint redeemAmount)
+    {
+        // Convert depositAmount to 18 decimals, which is required by Bancor formula
+        uint decimalConvertedDepositAmount = _convertAmountToRequiredDecimal(
+            _depositAmount, decimals(), eighteenDecimals
+        );
+        // Calculate redeem amount through bonding curve
+        redeemAmount = formula.calculateSaleReturn(
+            virtualTokenSupply,
+            virtualCollateralSupply,
+            reserveRatioForSelling,
+            decimalConvertedDepositAmount
+        );
+
+        // Convert redeem amount to collateral decimals
+        redeemAmount = _convertAmountToRequiredDecimal(
+            redeemAmount,
+            eighteenDecimals,
+            IERC20MetadataUpgradeable(address(token())).decimals()
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @dev Executes a buy order and updates the virtual supply of tokens and collateral.
+    /// This function internally calls `_buyOrder` to get the issuing amount and updates the
+    /// virtual balances accordingly.
+    /// @param _receiver The address of the recipient of the issued tokens.
+    /// @param _depositAmount The amount of collateral deposited for the buy order.
+    function _virtualBuyOrder(address _receiver, uint _depositAmount)
+        internal
+    {
+        uint amountIssued = _buyOrder(_receiver, _depositAmount);
+        _addVirtualTokenAmount(amountIssued);
+        _addVirtualCollateralAmount(_depositAmount);
```

Comment:
Our current implementation indeed sees the fee as part of the collateral. Because all the supplies are virtual anyway, this did not seem to be an issue and intended. Any thought here @0xNuggan?

For a different use case than the BCRG this indeed could lead to unwanted behaviour.

Reply:
Yes, we were treating it as part of the collateral, but you make a good point. I'll relay this to the BCRG so we can get their opinion

Reply:
After you're initial finding, and some discussions with the BCRG, we decided to follow your suggestion and exclude the fee from the balances